app/javascript/controllers/area_served_controller.js

```diff
@@ -0,0 +1,27 @@
+import { Controller } from "@hotwired/stimulus"
+export default class extends Controller {
+    static targets= ["share", "total", "warning"]
+    connect () {
+        this.calculateClientShareTotal()
+    }
+    calculateClientShareTotal ( ){
+        let total = 0;
+        this.share_targets.forEach(
```

Comment:
Sorry - this won't work :sweat\_smile: Stimulus requires a specific format for the target variables - it has to follow the JavaScript "CamelCase" format rather than the Ruby "snake\_case" format. So this has to read `this.shareTargets`.

Reply:
Hrm.  There's something more wrong than that.  I expect I mucked something up when I changed to a drop-down instead of numbers.

Reply:
By which I mean the tests aren't working.   It seems to be working on manual.